elytron/WFCORE-4842-tls-1.3-with-openssl.adoc

```diff
@@ -0,0 +1,121 @@
+provider with Elytron.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.redhat.com/browse/WFCORE-4842[WFCORE-4842]
+
+=== Related Issues
+
+* https://issues.jboss.org/browse/WFSSL-16[WFSSL-16]
+* https://issues.redhat.com/browse/WFLY-13085[WFLY-13085]
+* https://issues.jboss.org/browse/EAP7-1414[EAP7-1414]
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+* mailto:jstourac@redhat.com[Jan Stourac]
+
+=== Testing By
+// Put an x in the relevant field to indicate if testing will be done by Engineering or QE.
+// Discuss with QE during the Kickoff state to decide this
+
+[ ] Engineering
+
+[x] QE
+
+* TBD
+
+=== Affected Projects or Components
+
+* WildFly OpenSSL
+* WildFly Core
+* WildFly (documentation)
+
+=== Other Interested Projects
+
+== Requirements
+
+=== Hard Requirements
+
+* When using the OpenSSL TLS provider with Elytron, it should be possible to make use of TLS 1.3
+  by configuring the `cipher-suite-names` attribute in a `server-ssl-context` or in a
+  `client-ssl-context` in the Elytron subsystem.
+  ** Whether or not TLS 1.3 should be used by default when the OpenSSL TLS provider is in
+     use and the `cipher-suite-names` attribute has not been specified will need to be
+     determined using the performance tests that were used for
+     https://issues.jboss.org/browse/WFCORE-4172[WFCORE-4172].
+
+
+** Example configuration:
+
+[source,xml]
+----
+   /subsystem=elytron/server-ssl-context=myTLS13Context:add(cipher-suite-names=TLS_AES_128_CCM_8_SHA256:TLS_AES_256_GCM_SHA384, ...)
+----
+[source,xml]
+----
+   /subsystem=elytron/client-ssl-context=myTLS13ClientContext:add(cipher-suite-names=TLS_AES_128_CCM_8_SHA256:TLS_AES_256_GCM_SHA384:TLS_CHACHA20_POLY1305_SHA256:TLS_AES_128_GCM_SHA256, ...)
+----
+
+* When using the OpenSSL TLS provider with Elytron, it should be possible to make use of TLS 1.3
+  by configuring the `cipher-suite names` attriute in an `ssl-context` using the Elytron
+  authentication client.
+  ** Whether or not TLS 1.3 should be used by default when the OpenSSL TLS provider is in
+     use and the `cipher-suite names` attribute has not been specified will need to be
+     determined using the performance tests that were used for
+     https://issues.jboss.org/browse/WFCORE-4172[WFCORE-4172].
+
+** Example configuration:
+[source,xml]
+----
+   <ssl-context name="myTLS13Context">
+   ...
+     <cipher-suite names="TLS_AES_128_CCM_8_SHA256:TLS_AES_256_GCM_SHA384"/>
+   ...
+   </ssl-context>
+----
+
+=== Nice-to-Have Requirements
+
+=== Non-Requirements
+
+== Test Plan
+
+All testing should be done using the OpenSSL TLS provider. The TLS 1.3 cipher suites that are
+actually available will depend on this provider.
+
+One-way and two-way SSL tests will be added to the WildFly OpenSSL testsuite and to the WildFly
+Core testsuite that make use of the TLS 1.3 protocol and TLS 1.3 cipher suites. The `java.specification.version`
```

Comment:
Just checking whether this part is still true since we agreed that we'll cover this feature in one of our internal testsuites?

Reply:
Overall I don't believe we should ever take a decision to not include tests in the open source test suites if we have the option to perform the tests here regardless of any internal testing.

Ideally test cases should be as close to the engineer making changes as is practically possible so that any regressions can be picked up early - test cases within WildFly Core give us an opportunity to detect regressions definitely before WildFly Core is tagged and potentially before WildFly Open SSL is tagged.

Tests within the internal testsuite on the other hand only detect regressions late in the process potentially meaning any fix could require WildFly OpenSSL, WildFly Core, and WildFly all to be tagged again for something we had the potential to detect earlier.

Reply:
@jstourac Tests will still be added in the WildFly OpenSSL testsuite (I am currently working through these). Ideally, it would be good to have some test cases in the WildFly Core testsuite as well but I'll need to check with Ken first to see if we can perform the required setup on the Linux and Windows CI machines used for these runs.

Reply:
@darranl I completely agree. This case is a little bit more complicated with the setup necessary on platforms where OpenSSL libraries are not available at the moment in WildFly CI. That is why we talked with Farah that there will be tests for this in `WildFly OpenSSL upstream TS` but we're gonna use internal TS instead of `WildFly upstream TS` for integraiton tests. I wasn't quite sure what was the decision regarding to `WildFly Core upstream TS`.

@fjuma okay then, let me know in case you need anything. Thanks!